api-server/common/models/user.js

```diff
@@ -347,10 +348,14 @@ export default function(User) {
     if (!username && (!email || !isEmail(email))) {
       return Promise.resolve(false);
     }
-    log('checking existence');
-
-    // check to see if username is on blacklist
-    if (username && blacklistedUsernames.indexOf(username) !== -1) {
+    log('check if username is available');
+    // check to see if username is on blocklist
+    const usernameFilter = new badwordFilter();
+    if (
+      username &&
+      (blocklistedUsernames.indexOf(username) !== -1 ||
+        usernameFilter.isProfane(username) === true)
```

Comment:
```suggestion
      (blocklistedUsernames.includes(username) ||
        usernameFilter.isProfane(username))
```
Nitpicky, I know, but `isProfane` returns `true` or `false`.

Also, `isProfane` only catches things like `profanewordone-profanewordtwo` not `profanewordoneprofanewordtwo`.  Do we want to deal with those manually?

Reply:
> Do we want to deal with those manually?

Not sure how though, without doing so statically.